code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -81,11 +82,74 @@ func (msc *MinerSmartContract) GetNodepoolHandler(ctx context.Context, params ur
 }
 
 func (msc *MinerSmartContract) GetMinerListHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
-	allMinersList, err := msc.GetMinersList(balances)
+	var (
+		offsetString = params.Get("offset")
+		limitString  = params.Get("limit")
+		activeString = params.Get("active")
+		offset       = -1
+		limit        = -1
+		err          error
+	)
+	if offsetString != "" {
+		offset, err = strconv.Atoi(offsetString)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if limitString != "" {
+		limit, err = strconv.Atoi(limitString)
+		if err != nil {
+			return nil, err
+		}
+	}
+	filter := event.MinerQuery{}
+	if activeString != "" {
+		active, err := strconv.ParseBool(activeString)
+		if err != nil {
+			return nil, err
+		}
+		filter.Active = null.BoolFrom(active)
+	}
+
+	if balances.GetEventDB() == nil {
+		return nil, common.NewErrInternal("can't get miners. no db connection")
+	}
+
+	miners, err := balances.GetEventDB().GetMinersWithFiltersAndPagination(filter, offset, limit)
 	if err != nil {
 		return "", common.NewErrInternal("can't get miners list", err.Error())
 	}
-	return allMinersList, nil
+	minersArr := make([]interface{}, len(miners))
+	for i, miner := range miners {
```

Comment:
please use existng struct MinerNodes
we should not duplicate structs

Reply:
also we have function in minersc.minerTableToMinerNode() use transfoers functions

Reply:
The reason we are not using the existing struct is that the existing struct creates an error in json unmarshalling because there are 2 `id` in json `ID` and  `id` one is by the gorm and other by our event struct.

Reply:
thats why I said transform
[minersc.minerTableToMinerNode()](https://github.com/0chain/0chain/blob/0c0e7894394e469e150e29786aeced3119107d70/code/go/0chain.net/smartcontract/storagesc/eventdb.go#L19)